File: python/23matches.py
#!/usr/bin/python3
matches = 23
print(f'На столе лежит {matches} спичек.')
while matches > 0:
  move = 0
  while move < 1 or move > matches or move > 3:
    move = int(input('Сколько спичек вы возьмёте? '))
  matches -= move
  print(f'Вы взяли {move}, осталось {matches} спичек')
  if matches == 0:
    print('Последняя спичка ваша, я выиграл!')
    break
  move = (matches - 1) % 4
  if move == 0:
    move = 1
  matches -= move
  print(f'Я беру {move}, осталось {matches} спичек.')
  if matches == 0:
    print('Сегодня вам удалось победить. Посмотрим, что будет в следующий раз!')
    break

File: cpp/23matches.cpp
#include <iostream>

int main()
{
  int matches = 23;
  std::cout << "На столе лежит " << matches << " спичек.\n";
  while (matches > 0)
  {
    int move;
    do
    {
      std::cout << "Сколько спичек вы возьмёте? ";
      std::cin >> move;
    }
    while (move < 1 || move > matches || move > 3);
    matches -= move;
    std::cout << "Вы взяли " << move << ", осталось " << matches << " спичек.\n";
    if (matches == 0)
    {
      std::cout << "Последняя спичка ваша, я выиграл!\n";
      break;
    }
    move = (matches - 1) % 4;
    if (move == 0)
    {
      move = 1;
    }
    matches -= move;
    std::cout << "Я беру " << move << ", осталось " << matches << " спичек.\n";
    if (matches == 0)
    {
      std::cout << "Сегодня вам удалось победить. Посмотрим, что будет в следующий раз!\n";
      break;
    }
  }
}
